htdocs/comm/propal/list.php

```diff
@@ -1270,10 +1310,57 @@
 	$delallowed = $user->rights->propal->creer;
 
 	print $formfile->showdocuments('massfilesarea_proposals', '', $filedir, $urlsource, 0, $delallowed, '', 1, 1, 0, 48, 1, $param, $title, '', '', '', null, $hidegeneratedfilelistifempty);
+
+	if ($action == 'validate') {
+		if (GETPOST('confirm') == 'yes') {
+			$tmpproposal = new Propal($db);
+			$db->begin();
+			foreach ($toselect as $checked) {
+				if ($tmpproposal->fetch($checked)) {
+					if ($tmpproposal->statut == 0) {
+						if ($tmpproposal->valid($user)){
+							setEventMessage($tmpproposal->ref . " " . $langs->trans('PassedInOpenStatus'), 'mesgs');
+						} else {
+							setEventMessage($langs->trans('CantBeValidated'), 'errors');
+						}
+					} else {
+						setEventMessage($tmpproposal->ref . " " . $langs->trans('IsNotADraft'), 'errors');
+					}
+				}
+			}
+			$db->commit();
+		}
+	}
+
+	if ($action == "sign") {
+		if (GETPOST('confirm') == 'yes') {
+			$tmpproposal = new Propal($db);
+			$db->begin();
+			foreach ($toselect as $checked) {
+				$tmpproposal->fetch($checked);
+				if ($tmpproposal->fetch($checked)) {
+					if ($tmpproposal->statut == 1) {
+						$sqlp = "UPDATE " . MAIN_DB_PREFIX . "propal SET fk_statut = 2 WHERE rowid = " . $checked;
+						$resqlp = $db->query($sqlp);
+						if ($resqlp) {
+							setEventMessage($tmpproposal->ref . " " . $langs->trans('Signed'), 'mesgs');
+						} else {
+							dol_print_error($db);
+						}
+					} else {
+						setEventMessage($tmpproposal->ref . " " . $langs->trans('CantBeSign'), 'errors');
+					}
+				} else {
+					dol_print_error($db);
+				}
+			}
+			$db->commit();
```

Comment:
Add a test here if ($error) $db->rollback() else $db->commit()
$error must be set in the loop when you reach one error (when you have dol_print_error or setEventMessage with error.